app/build.gradle

```diff
@@ -20,14 +20,15 @@ android {
             storePassword 'release'
         }
     }
-    compileSdkVersion 22
-    buildToolsVersion "22.0.1"
+    compileSdkVersion 23
+    buildToolsVersion "23.0.2"
     defaultConfig {
         applicationId "net.pawelhajduk.daggerdemo"
         minSdkVersion 15
-        targetSdkVersion 22
+        targetSdkVersion 23
         versionCode 1
         versionName "1.0"
+        multiDexEnabled true
```

Comment:
However crazy it sounds: debug mode + new version of dagger and appcompat was causing the app to miserably crash on Nexus 5 ;-) Googling revealed that the problem is 64k ;-)

I guess it should be possible to enable multidex in debug and disable in prod?